File: src/__test__/cooker.test.js
import { Cooker } from "../model/Cooker";
import { Ingredient } from "../model/Ingredient";

describe("Cooker", () => {
  let cooker;
  let mockRecord;

  beforeEach(() => {
    cooker = new Cooker();
    mockRecord = jest.fn();
    mockRecord.mockReturnValue(1); 
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Initial cooker", () => {
    expect(cooker.waterLevel).toBe(0);
    expect(cooker.ingredientList).toHaveLength(0)
    expect(cooker.isButtonOn).toBeFalsy()
    expect(cooker.isPowerOn).toBeFalsy()
    expect(cooker.isCooking).toBeFalsy()
  });

  it("Cooking", async() => {
    let ingredient = new Ingredient("rice", 5)
    cooker.connectPower()
    await cooker.addWater(1)
    await cooker.addIngredient(ingredient)
    expect(cooker.waterLevel).toBe(1)
    expect(cooker.isPowerOn).toBe(true)
    await cooker.startCooking()
    expect(cooker.waterLevel).toBe(0)
  },20000)

});

File: test/cooker_test.dart
import 'package:test/test.dart';

import 'dart:async';

import '../model/cooker.dart';
import '../model/ingredient.dart';

void main() {
  late Cooker cooker;

  setUp(() {
    cooker = Cooker();
  });

  test('Initial cooker', () {
    expect(cooker.waterLevel, equals(0));
    expect(cooker.ingredientList.length, equals(0));
    expect(cooker.isButtonOn, equals(false));
    expect(cooker.isPowerOn, equals(false));
    expect(cooker.isCooking, equals(false));
  });

  test('Cooking', () async {
    final ingredient = Ingredient('rice', 5);
    cooker.connectPower();
    await cooker.addWater(1);
    await cooker.addIngredient(ingredient);
    expect(cooker.waterLevel, equals(1));
    expect(cooker.isPowerOn, equals(true));
    await cooker.startCooking();
    Timer(Duration(seconds: 1), () {
      expect(cooker.waterLevel, equals(0));
    });
  });
}
